test/test-cases/regression/variable-TIME_MON.json

```diff
@@ -33,7 +33,8 @@
       ]
     },
     "expected":{
-      "debug_log":"Target value: \"([0-9]+)\" \\(Variable: TIME_MON\\)"
+      "http_code": 200,
+      "debug_log":"Target value: \"[1-9][012]?\" \\(Variable: TIME_MON\\)"
```

Comment:
```suggestion
      "debug_log":"Target value: \"[1-9]|1[012]\" \\(Variable: TIME_MON\\)"
```

Reply:
Hmm... Your suggested pattern matches only with `"1`, the rest part is out of scope. I mean if the line contains this:
`"13"`
then the `"1` will match which is not enough for us.

Or em I wrong?

Reply:
You're right. It should be `^(?:1-9|1[012])$)`.

Reply:
I'm afraid we can't use anchors (`^` and `$`) (and probably your last suggestion contains a typo: the last `)` is unnecessary).

The test program compiles the regex and matches it with the whole log line.

Eg. with your suggestion the whole regex will be this:
```
Target value: \"^(?:1-9|1[012])$\" \\(Variable: TIME_MON\\)
```
and the subject is something like this (if it's correct):
```
Target value: "11" (Variable: TIME_MON)
```
But this does not match.

Reply:
Hm. But switching the order should do the trick: `1[012]|[0-9]`.

Reply:
No.

Subject:
```
Target value: "1" (Variable: TIME_MON)
```
does not match at all.
```
Target value: "10" (Variable: TIME_MON)
```
matches partially:
```
Target value: "10
```
What's wrong with the "original" one (which part of original PR)?

Reply:
I don't understand what the matcher is doing. Your original regex allows arbitrary numbers like `92`.

Reply:
TBH, I also don't know what that JSON thing does. If it's just a test, then maybe it doesn't matter and we'll just go with what you had originally.

Reply:
> I don't understand what the matcher is doing. Your original regex allows arbitrary numbers like `92`.

Ah, I see, you're right. As you figured out, the main goal is to cover the test as it accepts only numbers in range between 1 and 12. I'll try to figure out something.

Reply:
> TBH, I also don't know what that JSON thing does. If it's just a test, then maybe it doesn't matter and we'll just go with what you had originally.

Yes, it's just a test. Unfortunately there is no any solution to create a "dynamic" test, I mean use a request where the argument value is the current month... So we have to check the range in these cases. (Note, I think the current test is completely unusable.)